src/Twig/Extension/DynamicPropertyExtension.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Twig\Extension;
+
+use App\Form\Type\DynamicPropertiesType;
+use Symfony\Contracts\Translation\TranslatorInterface;
+use Twig\Extension\AbstractExtension;
+use Twig\TwigFunction;
+
+class DynamicPropertyExtension extends AbstractExtension
+{
+    private TranslatorInterface $translator;
+
+    public function __construct(TranslatorInterface $translator)
+    {
+        $this->translator = $translator;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getFunctions(): array
+    {
+        return [
+            new TwigFunction('dynamicPropertyValue', [$this, 'dynamicPropertyValue']),
+        ];
+    }
+
+    /**
+     * @param bool|string|int $value
+     */
+    public function dynamicPropertyValue($value, array $propertyDefinition): string
+    {
+        if (\in_array($propertyDefinition['type'], [DynamicPropertiesType::TYPE_CHOICE, DynamicPropertiesType::TYPE_CHOICE_WITH_OTHER], true)) {
+            return array_flip($propertyDefinition['choices'] ?? [])[$value] ?? $value;
+        }
+
+        if (DynamicPropertiesType::TYPE_BOOLEAN === $propertyDefinition['type']) {
+            return $this->translator->trans(sprintf('common.%s', (bool) $value ? 'yes' : 'no'));
+        }
+
+        if (DynamicPropertiesType::TYPE_NUMBER === $propertyDefinition['type']) {
+            return (string) $value;
+        }
+
+        $value = (string) $value;
+        if (\strlen($value) <= 75) {
+            return $value;
+        }
+
+        return substr($value, 0, 72).'...';
```

Comment:
imho it should be done in the twig template, with a title prop on the container

Reply:
OK, this way we can truncate differently depending on where the information is displayed